adhocteam/Head-Start-TTADP
# Add HSES username field to Users

**Description of change**

1) Add `hsesUsername` and `hsesAuthorities` fields to user model for reference
2) Make `email` an editable non-required field
3) Update admin ui to display hses readonly fields.
4) Save user name from hses when creating a new user.
5) audited migrations to ensure they were all returning Promises. This may do nothing, but might have fixed a weird backend test failure I was seeing where foreign keys were randomly not valid.

**How to test**

1) clear your db
2) log in via HSES to create new user and log out
3) log in as admin user
4) check new user created with additional fields.

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/291

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
